### ISSUE TITLE
[FR]: Change colours in 'Test filter' for readability and accessibility

### ISSUE THREAD

**Issue by Mitecon** [martinrotter/rssguard#982]
### Brief description of the feature request

Whenever I'm working on a filter and I click the 'Test' button, certain articles may be highlighted with a green or red background. When this is done, it's quite hard to check visually because the background, especially the green one does not contrast well with the text.

I want to be able to see that my filter did, indeed, correctly match the articles I hoped it would. I would like to be able to scan matching articles for false-positives.

The problem is that the text is white and the green background is very light:

![testing-filter-preview-green-is-hard-to-read](https://github.com/martinrotter/rssguard/assets/125831943/36b33841-ccfa-4bd4-9175-dbfeb06c29f1)

It's not so bad when the background is red:

![testing-filter-preview-red-is-not-so-bad](https://github.com/martinrotter/rssguard/assets/125831943/6da480de-741f-4de0-b88b-3a5754013bb8)

It's worth noting that I use a dark theme (Breath Dark on KDE Plasma) but I do not know of a way to adjust this just for RSS Guard.

The possibilities are to change the text to black whenever there is a green background. I'm not sure if that's feasible, though. The other possibility is to simply change from a bright green to a less bright green. But then that would hurt people who used light themes that have black text.

Basically just to change things so that there's a definite contrast between both text and coloured background.

However, depending upon dark/light themes and accessibility settings, this might not be ideal. On a dark theme, text is light and on a light theme, text is dark. So there's no one-size-fits-all setting here.

So I propose to offer a user-customisable colour picker.

**Comment by martinrotter** [comment]
Whole dialog was rewritten. Re-test and report back.
